src/contracts/Executor.sol

```diff
@@ -18,95 +22,130 @@ contract Executor is Context {
-        @notice If executeProposals fails it doesn't revert, emits {FailedHandlerExecution} event.
         @param proposals Array of Proposal which consists of:
         - originDomainID ID of chain deposit originated from.
         - resourceID ResourceID to be used when making deposits.
         - depositNonce ID of deposit generated by origin Bridge contract.
         - data Data originally provided when deposit was made.
         @notice Emits {ProposalExecution} event for each proposal in the batch.
-        @notice Behaviour: when execution fails, the handler will terminate the function with revert.
      */
-    function executeProposals(Proposal[] memory proposals) public whenBridgeNotPaused {
+    function executeProposals(
+        Proposal[] memory proposals,
+        bytes[] memory accountProof,
+        uint256 slot
+    ) public whenBridgeNotPaused {
         if (proposals.length == 0) revert EmptyProposalsArray();
 
         for (uint256 i = 0; i < proposals.length; i++) {
             if (isProposalExecuted(proposals[i].originDomainID, proposals[i].depositNonce)) {
                 continue;
             }
+            bytes32 stateRoot;
+            bytes32 storageRoot;
+            address routerAddress = _originDomainIDToRouter[proposals[i].originDomainID];
 
+            IStateRootStorage blockHeaderStorage = IStateRootStorage(_securityModels[proposals[i].securityModel]);
+            stateRoot = blockHeaderStorage.getStateRoots(proposals[i].originDomainID, slot);
+            storageRoot = StorageProof.getStorageRoot(accountProof, routerAddress, stateRoot);
             address handler = IBridge(_bridge)._resourceIDToHandlerAddress(proposals[i].resourceID);
-            bytes32 dataHash = keccak256(abi.encodePacked(handler, proposals[i].data));
-
             IHandler depositHandler = IHandler(handler);
+            verify(proposals[i], storageRoot);
 
             usedNonces[proposals[i].originDomainID][proposals[i].depositNonce / 256] |=
                 1 <<
                 (proposals[i].depositNonce % 256);
-            try depositHandler.executeProposal(proposals[i].resourceID, proposals[i].data) returns (
-                bytes memory handlerResponse
-            ) {
-                emit ProposalExecution(
-                    proposals[i].originDomainID,
-                    proposals[i].depositNonce,
-                    dataHash,
-                    handlerResponse
-                );
-            } catch (bytes memory lowLevelData) {
-                emit FailedHandlerExecution(lowLevelData, proposals[i].originDomainID, proposals[i].depositNonce);
-                usedNonces[proposals[i].originDomainID][proposals[i].depositNonce / 256] &= ~(1 <<
-                    (proposals[i].depositNonce % 256));
-                continue;
-            }
+            depositHandler.executeProposal(proposals[i].resourceID, proposals[i].data);
```

Comment:
Why don't we keep the try catch logic?
Don't we still need the usedNonces revert when a proposal execution fails.

Reply:
Why have we removed executeProposal bytes that were returned?

Reply:
A misunderstanding, understood we're removing it :see_no_evil: